exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,198 @@
+      console.log(rockQuestions.shift());
+    }
 
-      purses[currentPlayer] += 1
-      console.log(players[currentPlayer] + ' now has ' +
-          purses[currentPlayer] + ' Gold Coins.')
+  };
 
-      var winner = didPlayerWin()
+  this.playerIsInPenaltyBox = function () {
+    return inPenaltyBox[currentPlayer];
+  };
 
-      currentPlayer += 1
-      if (currentPlayer == players.length) { currentPlayer = 0 }
+  this.playerGetsOutOfPenaltyBox = function (roll) {
+    return roll % 2 != 0;
+  };
 
-      return winner
+  this.playerTakesTurn = function (roll) {
+    playersPositionOnTheBoard[currentPlayer] += roll;
+    if (
+      playersPositionOnTheBoard[currentPlayer] >
+      numberOfPositionsOnTheBoard - 1
+    ) {
+      playersPositionOnTheBoard[currentPlayer] -= numberOfPositionsOnTheBoard;
     }
-  }
+    console.log(
+      players[currentPlayer] + "'s new location is " +playersPositionOnTheBoard[currentPlayer]
+    );
+    console.log("The category is " + currentCategory());
+    askQuestion();
+  };
+
+
+  this.actionAfterRoll = function (roll) {
+    console.log(players[currentPlayer] + " is the current player");
+    console.log("They have rolled a " + roll);
+    if (this.playerIsInPenaltyBox() && this.playerGetsOutOfPenaltyBox(roll)) {
+      isGettingOutOfPenaltyBox = true;
+      console.log(
+        players[currentPlayer] + " is getting out of the penalty box"
+      );
+      this.playerTakesTurn(roll);
+    } else if (this.playerIsInPenaltyBox() & !this.playerGetsOutOfPenaltyBox(roll)) {
+      console.log( players[currentPlayer] + " is not getting out of the penalty box" );
+      isGettingOutOfPenaltyBox = false;
+    } else {
+      this.playerTakesTurn(roll);
+    }
+  };
 
-  this.wrongAnswer = function () {
-    console.log('Question was incorrectly answered')
-    console.log(players[currentPlayer] + ' was sent to the penalty box')
-    inPenaltyBox[currentPlayer] = true
 
-    currentPlayer += 1
-    if (currentPlayer == players.length) { currentPlayer = 0 }
-    return true
-  }
-}
+  this.moveToNextPlayer = function () {
+    currentPlayer += 1;
+    if (currentPlayer == players.length) {
+      currentPlayer = 0;
+    }
+  };
 
+  this.answerWasCorrect = function () {
+    if (inPenaltyBox[currentPlayer]) {
+      if (isGettingOutOfPenaltyBox) {
+        console.log("Answer was correct!!!!");
+        playersScore[currentPlayer] += 1;
+        console.log( players[currentPlayer] + " now has " +  playersScore[currentPlayer] + " Gold Coins." );
+        var winner = didPlayerWin();
+        this.moveToNextPlayer();
+        return winner;
+      } else {
+        this.moveToNextPlayer();
+        return true;
+      }
+    } else {
+      console.log("Answer was correct!!!!");
+      playersScore[currentPlayer] += 1;
+      console.log(players[currentPlayer] + " now has " + playersScore[currentPlayer] +" Gold Coins." );
+      var winner = didPlayerWin(); //if true, they don’t have 6 coins
```

Comment:
Lots of comments here, perhaps some renaming and refactoring could make the code more self explanatory